src/main/java/org/mockito/Mockito.java

```diff
@@ -1094,6 +1101,110 @@
  * verify(mock, times(2).description("someMethod should be called twice")).someMethod();
  * </code></pre>
  *
+ * <h3 id="36">36. <a class="meaningful_link" href="#Java_8_Lambda_Matching">Java 8 Lambda Matcher Support</a> (Since 2.0.0)</h3>
+ * <p>
+ * You can use Java 8 lambda expressions with {@link ArgumentMatcher} to reduce the dependency on {@link ArgumentCaptor}
+ * The argument will be passed to the ArgumentMatcher as a strongly typed object, so it is possible
+ * to do anything with it.
+ * <p>
+ * Examples:
+ * <p>
+ * <pre class="code"><code class="java">
+ *
+ * // verify a list only had strings of a certain length added to it
+ * // note - this will only compile under Java 8
+ * verify(list, times(2)).add(string -> string.length() < 5);
+ *
+ * // Java 7 equivalent
+ * verify(list, times(2)).add(new ArgumentMatcher<String>(){
+ *     boolean matches(String arg) {
+ *         return arg.length() < 5;
+ *     }
+ * });
+ *
+ * // more complex Java 8 example - where you can specify complex verification behaviour functionally
+ * verify(target, times(1)).receiveComplexObject(obj -> obj.getSubObject().get(0).equals("expected"));
+ * </code></pre>
+ *
+ * <h3 id="37">37. <a class="meaningful_link" href="#Java_8_Custom_Answers">Java 8 Custom Answer Support</a> (Since 2.0.0)</h3>
+ * <p>
+ * As the {@link Answer} interface has just one method it is already possible to implement it in Java 8 using
+ * a lambda expression for very simple situations. The more you need to use the parameters of the method call,
+ * the more you need to typecast the arguments from {@link org.mockito.invocation.InvocationOnMock}.
+ *
+ * <p>
+ * Examples:
+ * <p>
+ * <pre class="code"><code class="java">
+ * // answer by returning 12 every time
+ * doAnswer(invocation -> {return 12;})
+ *     .when(mock).doSomething();
+ *
+ * // answer by using one of the parameters - converting into the right
+ * // type as your go - in this case, returning the length of a string parameter
+ * // as the answer
+ * doAnswer(invocation -> invocation.getArgumentAt(2, String.class).length())
```

Comment:
If I am not mistaken, `String.class` can be omitted here: http://site.mockito.org/mockito/docs/current/org/mockito/invocation/InvocationOnMock.html#getArgument(int)